include/plugins/slotproxy.h

```diff
@@ -75,18 +76,25 @@ class SlotProxy
      */
     constexpr auto generate(auto&&... args)
     {
-        if (plugin_.has_value())
+        if (! plugins_.empty())
         {
-            return plugin_.value().generate(std::forward<decltype(args)>(args)...);
+            return plugins_.front().generate(std::forward<decltype(args)>(args)...);
         }
         return std::invoke(default_process, std::forward<decltype(args)>(args)...);
     }
 
     constexpr auto modify(auto& original_value, auto&&... args)
     {
-        if (plugin_.has_value())
+        if (! plugins_.empty())
         {
-            return plugin_.value().modify(original_value, std::forward<decltype(args)>(args)...);
+            auto modified_value = original_value;
```

Comment:
I think the efficiency here can be improved a bit by making this a reference (or maybe rename `original_value`), since I don't think we need the original `original_value` for anything after we're done (or during).